src/Workspaces/Core/Portable/Formatting/Engine/TokenStream.cs

```diff
@@ -271,28 +269,50 @@ public int GetCurrentColumn(SyntaxToken token)
 
         public int GetCurrentColumn(TokenData tokenData)
         {
-            return GetColumn(tokenData, _getTriviaData);
+            return GetColumn(tokenData, new TriviaDataGetter(this));
         }
 
         public int GetOriginalColumn(SyntaxToken token)
         {
             var tokenWithIndex = this.GetTokenData(token);
-            return GetColumn(tokenWithIndex, _getOriginalTriviaData);
+            return GetColumn(tokenWithIndex, new OriginalTriviaDataGetter(this));
+        }
+
+        interface ITriviaDataGetter
+        {
+            TriviaData GetTriviaData(TokenData token1, TokenData token2);
+        }
+
+        struct TriviaDataGetter : ITriviaDataGetter
+        {
+            readonly TokenStream _tokenStream;
+            public TriviaDataGetter(TokenStream tokenStream) { _tokenStream = tokenStream; }
+
+            public TriviaData GetTriviaData(TokenData token1, TokenData token2) => _tokenStream.GetTriviaData(token1, token2);
+        }
+
+        struct OriginalTriviaDataGetter : ITriviaDataGetter
+        {
+            readonly TokenStream _tokenStream;
+            public OriginalTriviaDataGetter(TokenStream tokenStream) { _tokenStream = tokenStream; }
+
+            public TriviaData GetTriviaData(TokenData token1, TokenData token2) => _tokenStream.GetOriginalTriviaData(token1, token2);
         }
 
         /// <summary>
         /// Get column of the token 
         /// * column means text position on a line where all tabs are converted to spaces that first position on a line becomes 0
         /// </summary>
-        private int GetColumn(TokenData tokenData, Func<TokenData, TokenData, TriviaData> triviaDataGetter)
+        private int GetColumn<TDataGetter>(TokenData tokenData, TDataGetter dataGetter)
+            where TDataGetter : ITriviaDataGetter
```

Comment:
can you add the 'struct' constraint.  it will help indicate that there shouldn't be boxing here.

Reply:
Added the constraint